core/src/main/scala/kafka/server/epoch/LeaderEpochFileCache.scala

```diff
@@ -92,17 +92,13 @@ class LeaderEpochFileCache(topicPartition: TopicPartition, leo: () => LogOffsetM
     * if requestedEpoch is < the first epoch cached, UNSUPPORTED_EPOCH_OFFSET will be returned
     * so that the follower falls back to High Water Mark.
     *
-    * @param requestedEpoch requested leader epoch
+    * @param requestedEpoch requested leader epoch. Must be non-negative
```

Comment:
One use which seems unprotected is `Partition.lastOffsetForLeaderEpoch`. Seems like it will take a bit more work to purge the use of UNDEFINED_EPOCH.. Unless you're up for following through the full usage, maybe we had better leave the check for UNDEFINED_EPOCH in place.

Reply:
We did 70% of the work, might as well do the last 30%. As far as I can tell, they're only 4 calls to that method (3 of which from tests), all go through `ReplicaManager#lastOffsetForLeaderEpoch`

Reply:
Can we raise an exception if the epoch is non-negative? Also, I think the docstring above needs to be updated.